Quorum/apis/llms/llm_conversation.py

```diff
@@ -0,0 +1,105 @@
+from enum import StrEnum, auto
+from pathlib import Path
+
+class MessageRole(StrEnum):
+    USER = auto()
+    ASSISTANT = auto()
+
+
+MAX_TOKENS = 1024
+
+
+def tmp_send_message_to_claude(client, model, messages):
+    try:
+        response = client.messages.create(
+            max_tokens=MAX_TOKENS,
+            model=model,
+            messages=messages
+        )
+    except:
+        raise
+
+    with open(Path(__file__).parent / 'ClaudeExecutionsNum.txt', 'r+') as f:
+        num = int(f.read())
+        f.seek(0)
+        f.write(str(num + 1))
+        f.truncate()
+
+    return response
+
+
+class LLMConversation:
+    '''
+    Description
+    -----------
+    At least GPT and Claude share the exact same API. Probably other LLMs use the same API as well,
+    just using different clients. This abstract class builds is expected to be inherited and build
+    the relevant client to be used with the same implementation of interacting with the LLM while 
+    maintaining a conversation context.
+    '''
+    def __init__(self):
+        '''
+        Summary
+        -------
+        Initialize a conversation manager for an LLM.
+        '''
+        self.client = None
+        self.model = None
+        self.messages: list[dict[str, str]] = []
+        
+    def add_message(self, role: str, content: str) -> None:
+        '''
+        Summary
+        -------
+        Add a message to the conversation history.
+        
+        Parameters
+        ----------
+        role : MessageRole
+            What role to use.
+        content : str
+            The message content.
+        '''
+        self.messages.append({
+            'role': str(role),
+            'content': content
+        })
+    
+    def send_message(self, message: str) -> str:
+        '''
+        Summary
+        -------
+        Send a message to the LLM and get the response, maintaining conversation history.
+        
+        Parameters
+        ----------
+        message : str
+            The message to send.
+            
+        Returns
+        -------
+        str
+            The response.
+        '''
+        # Add user message to history
+        self.add_message(MessageRole.USER, message)
+        
+        # Get response from LLM
+        response = tmp_send_message_to_claude(self.client, self.model, self.messages)
+        
+        # Add LLM's response to history
+        assistant_message = response.content[0].text
```

Comment:
Why content[0]? can you add description?

Reply:
From the example I created to show the message chaining. Here's the message object the LLM returns:
```
Message(id='msg_012n4J5ULfTKmbrzBC1iqUNS', content=[TextBlock(text=', happy to help you with anything you need. Feel free to ask me a question or let me know what I can do for you.', type='text')], model='claude-3-5-haiku-20241022', role='assistant', stop_reason='end_turn', stop_sequence=None, type='message', usage=Usage(input_tokens=13, output_tokens=31))
```